linked_list/main.py

```diff
@@ -42,54 +42,103 @@ class Node(object):
     """
     Node class representing each of the linked nodes in the list.
     """
-
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        return isinstance(other, Node) and self.elem == other.elem
 
     def __repr__(self):
-        pass
+        return "Node: {0} with next node: {1}".format(self.elem, self.next)
 
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+            
+        if elements:
+            for element in elements:
+                self.append(element)
```

Comment:
Nice usage of the append() in the __init__ , much more efficient than what we were trying to come up with initially.